eppoclient/configresponse.go

```diff
@@ -84,35 +90,40 @@ func (v *variationType) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-func (ty variationType) valueToAssignmentValue(value interface{}) (interface{}, error) {
+func (ty variationType) valueToAssignmentValue(value interface{}, unmarshalJson bool) (AssignmentValue, error) {
 	switch ty {
 	case stringVariation:
 		s := value.(string)
-		return s, nil
+		return AssignmentValue{Parsed: s, Raw: []byte(s)}, nil
 	case integerVariation:
 		f64 := value.(float64)
 		i64 := int64(f64)
 		if f64 == float64(i64) {
-			return i64, nil
+			return AssignmentValue{Parsed: i64, Raw: []byte(strconv.FormatInt(i64, 10))}, nil
 		} else {
-			return nil, fmt.Errorf("failed to convert number to integer")
+			return AssignmentValue{}, fmt.Errorf("failed to convert number to integer")
 		}
 	case numericVariation:
 		number := value.(float64)
-		return number, nil
+		return AssignmentValue{Parsed: number, Raw: []byte(strconv.FormatFloat(number, 'f', -1, 64))}, nil
```

Comment:
Float formatting isn't exactly cheap. Although Go uses a pretty fast float to string algorithm (ryu), it's not free.

This serialization is pure waste given the raw value is discarded